pkg/util/testingjobs/mpijob/wrappers_mpijob.go

```diff
@@ -42,44 +42,88 @@ func MakeMPIJob(name, ns string) *MPIJobWrapper {
 			RunPolicy: kubeflow.RunPolicy{
 				Suspend: ptr.To(true),
 			},
-			MPIReplicaSpecs: map[kubeflow.MPIReplicaType]*kubeflow.ReplicaSpec{
-				kubeflow.MPIReplicaTypeLauncher: {
-					Replicas: ptr.To[int32](1),
-					Template: corev1.PodTemplateSpec{
-						Spec: corev1.PodSpec{
-							RestartPolicy: corev1.RestartPolicyNever,
-							Containers: []corev1.Container{
-								{
-									Name:      "c",
-									Image:     "pause",
-									Command:   []string{},
-									Resources: corev1.ResourceRequirements{Requests: corev1.ResourceList{}},
-								},
-							},
-							NodeSelector: map[string]string{},
-						},
+			MPIReplicaSpecs: make(map[kubeflow.MPIReplicaType]*kubeflow.ReplicaSpec),
+		},
+	},
+	}
+}
+
+type MPIJobReplicaSpecRequirement struct {
+	ReplicaType   kubeflow.MPIReplicaType
+	Name          string
+	ReplicaCount  int32
+	Annotations   map[string]string
+	RestartPolicy kubeflow.RestartPolicy
+}
+
+func (j *MPIJobWrapper) MPIJobReplicaSpecs(replicaSpecs ...MPIJobReplicaSpecRequirement) *MPIJobWrapper {
+	j = j.MPIJobReplicaSpecsDefault()
+	for _, rs := range replicaSpecs {
+		j.Spec.MPIReplicaSpecs[rs.ReplicaType].Replicas = ptr.To[int32](rs.ReplicaCount)
+		j.Spec.MPIReplicaSpecs[rs.ReplicaType].Template.Name = rs.Name
```

Comment:
Not sure that we need to set template.Name. Maybe like this:

```suggestion
		j.Spec.MPIReplicaSpecs[rs.ReplicaType].Template.Spec.Containers[0].Name = rs.Name
```

?

Reply:
those are different names, one is for the job itself and other for containers, which should be equal certain name - 
[The pod must include a container named tensorflow.](https://www.kubeflow.org/docs/components/training/user-guides/tensorflow/)